kotest-property/src/commonMain/kotlin/io/kotest/property/context.kt

```diff
@@ -25,6 +27,32 @@ class PropertyContext(val config: PropTestConfig = PropTestConfig()) {
       failures++
    }
 
+   /**
+    * Setup this [PropertyContext] for a single run of property test.
+    *
+    * This function sets the [contextualRandomSource] as well as clears the [generatedSamples].
+    */
+   internal fun setupContextual(rs: RandomSource) {
+      contextualRandomSource = rs
+      generatedSamples.clear()
+   }
+
+   internal fun generatedSamples(): List<Sample<*>> = generatedSamples
+
+   /**
+    * Returns a [RandomSource] that can be used within this [PropertyContext]
+    */
+   fun randomSource(): RandomSource = contextualRandomSource ?: RandomSource.default()
+
+   /**
+    * Extracts a sample or edgecase value from an [Arb] using the contextual [RandomSource].
```

Comment:
@sksamuel I'm trying to find the correct place to put this under the web documentation. Do you have a suggestion?